GoogleForCreators/web-stories-wp
# Dashboard: remove `canReadPrivatePosts` capability check

## Context



## Summary

Previously the number of private story was leaked in the API, to not show this value, it has a permission check in the client. This change moves the check for the rest api.

It also refactors logic, to make it a little cleaner and allows for but scheduled story. 

## Relevant Technical Choices



## To-do



## User-facing changes

### Logged is as an admin

<img width="1299" alt="Screenshot 2021-08-06 at 15 03 24" src="https://user-images.githubusercontent.com/237508/128522317-07b3ad78-2a5f-4da7-a22e-4a246a17e3e2.png">

### Logged in as contributor

<img width="1265" alt="Screenshot 2021-08-06 at 15 36 38" src="https://user-images.githubusercontent.com/237508/128527106-99d4a3ae-574b-4362-b5d7-70bde4d30faf.png">


## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Create a number of draft, published, scheduled and private stories. 
2. Go to dashboard ( logged in as admin ). See filters. 
3. Go login as contributor. 
2. Go to dashboard.  See filters, notice that private filter should not be present. 

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #
